external-resolver/src/test/scala/org/genivi/sota/resolver/test/random/Misc.scala

```diff
@@ -32,6 +37,15 @@ object Misc {
 
       def flatMap[A, B](fa: () => A)(f: A => () => B): () => B =
         () => f(fa())()
+
+      override def tailRecM[A, B](a: A)(f: (A) => () => Either[A, B]): () => B =
```

Comment:
Could we have a `@tailrec` annotation on this method. You need to import `scala.annotation.tailrec` to enable.